### ISSUE TITLE
`aiidalab/full-stack:edge` container crashing

### ISSUE THREAD

**Issue by edan-bainglass** [aiidalab/aiidalab-docker-stack#494]
Crashing with the following message:
```python
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/notebook/traittypes.py", line 235, in _resolve_classes
    klass = self._resolve_string(klass)
  File "/home/jovyan/.local/lib/python3.9/site-packages/traitlets/traitlets.py", line 2015, in _resolve_string
    return import_item(string)
  File "/home/jovyan/.local/lib/python3.9/site-packages/traitlets/utils/importstring.py", line 33, in import_item
    module = __import__(package, fromlist=[obj])
ModuleNotFoundError: No module named 'jupyter_server.contents'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/bin/jupyter-notebook", line 10, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.9/site-packages/jupyter_core/application.py", line 269, in launch_instance
    return super().launch_instance(argv=argv, **kwargs)
  File "/home/jovyan/.local/lib/python3.9/site-packages/traitlets/config/application.py", line 1073, in launch_instance
    app = cls.instance(**kwargs)
  File "/home/jovyan/.local/lib/python3.9/site-packages/traitlets/config/configurable.py", line 583, in instance
    inst = cls(*args, **kwargs)
  File "/home/jovyan/.local/lib/python3.9/site-packages/traitlets/traitlets.py", line 1292, in __new__
    inst.setup_instance(*args, **kwargs)
  File "/home/jovyan/.local/lib/python3.9/site-packages/traitlets/traitlets.py", line 1335, in setup_instance
    super(HasTraits, self).setup_instance(*args, **kwargs)
  File "/home/jovyan/.local/lib/python3.9/site-packages/traitlets/traitlets.py", line 1311, in setup_instance
    init(self)
  File "/opt/conda/lib/python3.9/site-packages/notebook/traittypes.py", line 226, in instance_init
    self._resolve_classes()
  File "/opt/conda/lib/python3.9/site-packages/notebook/traittypes.py", line 238, in _resolve_classes
    warn(f"{klass} is not importable. Is it installed?", ImportWarning)
TypeError: warn() missing 1 required keyword-only argument: 'stacklevel'
```

**Comment by danielhollas** [comment]
@edan-bainglass thanks for the report. Is it crashing on container startup? Where exactly do you see this trace?

**Comment by edan-bainglass** [comment]
On container startup

**Comment by unkcpz** [comment]
Just provide more context, if I understand correctly from the discussion with @edan-bainglass, he changed to use `edge` image for his profile and it can not start with the exception above. My guess was some jupyter core package installed in his home folder that cause the conflict. I remember @danielhollas pinned a jupyter package version could that be the cause of the issue?

**Comment by danielhollas** [comment]
>  jupyter package version could that be the cause of the issue?

No, I don't think that's the issue. But I believe I saw something similar when i was upgrading the container to the edge image. 

@edan-bainglass Can you provide the output of `pip list --user` in the container?
You're going to have to run the container manually with attached volume, depending on your config, something like (substitute `<profile_name>` with the name of your profile if you use aiidalab-launch

```
docker run -v aiidalab_<profile_name>_home:/home/jovyan ghcr.io/aiidalab/full-stack:edge pip list --user
```

**Comment by danielhollas** [comment]
> My guess was some jupyter core package installed in his home folder that cause the conflict.

Yes, I'd guess this is likely. Another thing to know is what was the previous version of the image that worked?

**Comment by danielhollas** [comment]
According to this issue, downgrading or upgrading traitlets might fix this https://github.com/jupyter/notebook/issues/7048

**Comment by danielhollas** [comment]
It seems that this is caused by `traitlets>=5.10.0` . @edan-bainglass can you please confirm what version of traitlets are you running? You should be able to get it working by installing `traitlets==5.9.0`. 

I would also be interested about which version of the image have you been running previously. AFAIK this is not a problem with `edge`, I see in with `latest` and also with previous versions, so now I am thoroughly confused as to how this ever worked and why we did not notice it previously???

Ohhh, I found it. :sweat_smile: It looks like this issue has been introduced only very recently in `aiidalab-widgets-base==2.3.0a1` where I unpinned the traitlets version specifier from `==5.9.0` to `~=5.9`, I didn't know that newer versions did not work. I'll put up some PRs to fix this, but in the meantime @edan-bainglass you'll have to reinstall traitlets manually.

**Comment by edan-bainglass** [comment]
@danielhollas thanks for taking care of this. Note that it is no longer a concern for me at the moment. 

A bit of context - I was exploring [Solara](https://solara.dev/) for app/widget development, specifically for the `QueryBuilder`-widget. As it would be part of the widgets base repo, I figured I'd develop in an AiiDAlab container. Solara requires ipywidgets v8.x. Hence, I checked out the `full-stack:edge` image. Then the above issue occurred. However, in the meantime, I had decided to instead develop in a separate/dedicated repo. Therefore, no longer an issue.

In any case, good we caught this bug. Thanks again @danielhollas, @unkcpz  for addressing it.
